js/tweets.js

```diff
@@ -87,97 +87,98 @@ See documentation at https://github.com/fossasia/fossasia-loklak-webtweets
-                                          // as 2nd argument
-        throw new Error('[LOKLAK-FETCHER] No callback provided');
-      }
-
-      var settings = [ 'count', 'source', 'tzOffset', 'minified' ];  // Field names for all the possible parameters
-      var defaults = [ 100, 'all', new Date().getTimezoneOffset(), true ];  // Default values
-
-      // Check if no options have been provided
-      if(typeof options === 'undefined') {
-        var options = {}; // Create 'options' to avoid ReferenceErrors later
-      }
-      var query = "";
-      //Check if there are any data elements set
-      var tweetsEl = document.getElementsByClassName("tweets-feed")[0];
-      var dataset = tweetsEl.dataset;
-      if(dataset.count) {
-        options[settings[0]] = dataset.count; //count is index 0
-      }
-
-      if(dataset.query) {
-        query = dataset.query.replace(/\s/gi, '%20').replace(/#/gi, '%23'); //replace spaces and hashtags in URL
-      } else {
-        query = "";
-      }
-
-      if(dataset.start) {
-        query = query + "%20since:" + dataset.start;
-      }
-
-      if(dataset.end) {
-        query = query + "%20until:" + dataset.end;
-      }
-
-      if(dataset.from) {
-        var sources = dataset.from.split(',');
-	sources.forEach(source => {
-	  query = query + "from:" + source + "%20OR%20"
-	})
-	query = query.substring(0, query.length - 8);
-      }
-
-      // Write unset options as their default
-      for(index in settings) {
-        if(options[settings[index]] === undefined) {
-          options[settings[index]] = defaults[index];
+    var script = null;
+
+    loklakFetcher = {
+        /**
+         * Fetches tweets from the public loklak API, with the options provided
+         * @param  {object}   options  Object with allowed GET-attributes, see
+         *                             loklak.org/api.html
+         * @param  {function} callback Function called after getting the results.
+         *                             These are passed as first argument
+         */
+        getTweets: function(options, callback) {
+            if(typeof options === 'function') { // A callback has been provided as 2nd
+                // argument (no options)
+                var callback = options;
+                options = {};
+            } else if(callback === undefined) { // No callback has been provided, even
+                // as 2nd argument
+                throw new Error('[LOKLAK-FETCHER] No callback provided');
+            }
+
+            var settings = [ 'count', 'source', 'fields', 'limit', 'tzOffset',
+                'minified' ];  // Field names for all the possible parameters
+            var defaults = [ 100, 'cache', '', '', 0, true ];  // Default values
+
+            // Check if no options have been provided
+            if(typeof options === 'undefined') {
+                var options = {}; // Create 'options' to avoid ReferenceErrors later
+            }
+
```

Comment:
Add ```query``` declaration as it was, it can cause errors in scope .

Reply:
please see my above comment. And thanks a lot for reviewing

Reply:
>Add query declaration as it was, it can cause errors in scope .

@akshatnitd , if you take a close look then you can see that the `query` declaration is the same. It's only that github is showing differences due to indentation. The code is the same for the query part. The scope remain unchanged.

Reply:
@himanish-star I can see that you have defined ```var query``` in line 127 inside the `if` statement, you should let the declaration be at line 117, as it was before. It can give an error of ```query not defined``` when line 129 gets executed.